libs/wingc/src/type_check.rs

```diff
@@ -5257,6 +5408,10 @@ where
 {
 	let (message, span) = match lookup_result {
 		LookupResult::NotFound(s) => (format!("Unknown symbol \"{s}\""), s.span()),
+		LookupResult::MultipleFound => (
+			format!("Ambiguous symbol \"{looked_up_object}\""),
+			looked_up_object.span(),
+		),
```

Comment:
I don't think this error should appear, but I couldn't really be sure so I decided to leave it here just in case

Reply:
Since the symbol environment is a map, I don't think it is possible to find multiple symbols with the same name within an environment, no?

Reply:
Good question. It is a map, but the entries of the map aren't always individual types - they can also refer to namespaces (see the `SymbolKind` enum). In this PR I've updated namespaces so they now can point to multiple environments (to support looking up a symbol in a directory), hence why it's now possible to have multiple matches during a symbol lookup.

Reply:
Can't we design it in such a way that if a dir has multiple files in it and none of them are explicitly brought then the dir's namespace will have a single env aggregating all the types in all the files in that level of the dir?
Bringing one of the files explicitly will by definition create another namespace where we'll just add those types into again.

Reply:
I didn't have a chance to try that approach, though it sounds like it could also work. I think my only concern is that it would require copying all of the symbols from the other symbol environments into a new symbol environment (unless there is another way...?) -- and since that would mean there's no longer one source of truth for a given symbol (as it might be defined in multiple places), it could lead to bugs in the future. For example, maybe there could be issues with only a subset of files in the project are re-type-checked by the LSP for example (incremental compilation). But I'm not sure.